table/sst_file_dumper.cc

```diff
@@ -125,7 +125,7 @@ Status SstFileDumper::GetTableReader(const std::string& file_path) {
                               nullptr);
       file_.reset(new RandomAccessFileReader(std::move(file), file_path));
     }
-    options_.comparator = &internal_comparator_;
+    options_.comparator = internal_comparator_.user_comparator();
```

Comment:
where is the `options_.comparator ` referenced? Is it passed into table reader or sst file reader?

Reply:
It's like not passed. The subsequent `NewTableReader` call uses `ioptions_` which is constructed from `options_` and has a `comparator` member.

Reply:
Right. Therefore, I suspect it's not at all used. We should fix this, or at least make it clearer.

Reply:
@satyajanga to clarify, this PR is only intended for testing and I currently do not plan to merge it.
Feel free to discuss and address this issue in your PR.